### PR TITLE
FIREFLY-1480: Fix binned plot and chart-saving bugs

### PR DESCRIPTION
**Pull request by jaladh-singhal**
Fixes [FIREFLY-1480](https://jira.ipac.caltech.edu/browse/FIREFLY-1480)

Three bugs in ticket:
1. no longer reproducible
2. when an expression is given to X, Y to heatmap options it was closing the dialog after "apply". It was requesting table stats even when they were already available - which takes time and was causing isNonNumColumn() to throw errors.
3. Plot saving wasn't working because CompleteButton wasn't connected to FieldGroup (this got taken out during joy-ui refactor)

## Testing
https://fireflydev.ipac.caltech.edu/firefly-1480-binned-plot-bugs/firefly/

Select 2MASS catalog, do m81, 1deg -> open heatmap options -> change x: j_m-h_m, y: h_m-k_m -> apply. Chart should update without closing. 

[For regression testing] Do another search, 2MASS, m81, 1arcmin -> open scatter options -> change x,y as above -> apply. Chart should behave like before.

Click on plot save button, and click save - it should download a png for chart.

### CODE REVIEW

File: src/firefly/js/charts/ChartUtil.js [hunk 1]
```diff
@@ -578,7 +578,9 @@ function updateChartData(chartId, traceNum, tablesource, action={}) {
     } else {
         if (!isFullyLoaded(tbl_id)) return;
         const tableModel = getTblById(tbl_id);
-        dispatchLoadTblStats(tableModel.request);
+        if (!getColValStats(tbl_id)) {
+            dispatchLoadTblStats(tableModel.request);
+        }
```
Comment (loitly) [inline src/firefly/js/charts/ChartUtil.js:583]: Good catch.

File: src/firefly/js/charts/ui/PlotlySaveDialog.jsx [hunk 2]
```diff
@@ -63,25 +64,24 @@ async function saveFile(request, Plotly, chartDiv) {
...
                         label: 'Chart Filename',
                     }} />
-            </FieldGroup>
-
-            <Stack flexDirection='row' justifyContent='space-between'>
-                <Stack spacing={1} direction='row' alignItems='center'>
-                    <CompleteButton text='Save' dialogId={DIALOG_ID}
-                                    onSuccess={(request) => saveFile(request,Plotly, chartDiv)} />
-                    <Button onClick={() => dispatchHideDialog(DIALOG_ID)}>Cancel</Button>
+                <Stack flexDirection='row' justifyContent='space-between'>
```
Comment (loitly) [inline src/firefly/js/charts/ui/PlotlySaveDialog.jsx:76]: Can you change `flexDirection` to `direction`.  I know they are everywhere, but let's fix them when we encounter them.

**Review by loitly — APPROVED** [review]
Tested, changes look good.

**Comment by lrebull** [comment]
Yep, looks good.

File: src/firefly/js/charts/ui/PlotlySaveDialog.jsx (hunk 2 above)
Comment (jaladh-singhal) [reply src/firefly/js/charts/ui/PlotlySaveDialog.jsx:76]: sure - overlooked it!

**Pull request merged**